ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/DiffExtensions.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.infrastructure
+
+import ai.x.diff.{Comparison, DiffShow, Different, Identical, green, red}
+import scalaz.{@@, Tag}
+
+trait DiffExtensions {
```

Comment:
In the spirit of _composition-over-inheritance_ I guess this can be made into an object as well. :slightly_smiling_face:

Reply:
Unfortunately, because `ai.x.diff` has a base implicit conversion for any `T` that simply throws an error, it gives us runtime exceptions instead of compilation errors. (I cannot figure out why.)

In case you're wondering, it looks like this:

```scala
  implicit def otherDiffShow[T: scala.reflect.ClassTag]: DiffShow[T] = new DiffShow[T] {
    private val T = scala.reflect.classTag[T].toString
    // throw new Exception( s"Cannot find DiffShow[$T]" )
    def show( v: T ) = red( new Exception( s"ERROR: Cannot find DiffShow[$T] to show value " + v ).showStackTrace )
    def diff( l: T, r: T ) = Error( new Exception( s"ERROR: Cannot find DiffShow[$T] to show values ($l, $r)" ).showStackTrace )
  }
```

This means that if we turn this into an object and import it where required, forgetting the import doesn't lead to a compilation error but an error in your tests, which means that you may not notice for a while. This is less than fun. For this reason, I suggest we keep it like this unless it starts to blow up.